zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/CommandDistributionAcknowledgedApplier.java

```diff
@@ -23,6 +23,12 @@ public CommandDistributionAcknowledgedApplier(final MutableDistributionState dis
 
   @Override
   public void applyState(final long key, final CommandDistributionRecord value) {
-    distributionState.removePendingDistribution(key, value.getPartitionId());
+    final var partitionId = value.getPartitionId();
+
+    distributionState
+        .queueForDistribution(key)
+        .ifPresent(queue -> distributionState.popQueuedDistribution(queue, partitionId));
```

Comment:
❌ If you applied the other changes I suggested I'd propose having a `removeQueuedDistribution` method instead of a pop. You have the `distributionKey` available here so we can do a targeted removal. Currently this works by assumption that the acknowledgement relates to the first distribution in the queue. I suppose it works, but if something changes and we start popping the wrong queued distribution it's gonna be very difficult to rootcause the bug.